### PR TITLE
SiteService:Applications Endpoints

### PR DESCRIPTION
**Pull request by srachakonda**

### CODE REVIEW

File: src/main/scala/com/imaginea/Main.scala [hunk 1]
```diff
@@ -281,7 +281,48 @@ object Main extends App {
...
   }
 
-  implicit val site1Format = jsonFormat(Site1.apply, "id", "siteName", "instances", "reservedInstanceDetails", "filters", "loadBalancers", "scalingGroups", "groupsList", "groupBy")
+
+  implicit object SiteFormat extends RootJsonFormat[Site1] {
+    override def write(i: Site1): JsValue = {
+      val fieldNames = List("id", "siteName", "instances", "reservedInstanceDetails", "filters",
+        "loadBalancers", "scalingGroups", "groupsList", "applications", "groupBy")
+      // scalastyle:off magic.number
+      val fields = new collection.mutable.ListBuffer[(String, JsValue)]
```
Comment (eklavya) [inline src/main/scala/com/imaginea/Main.scala]: why not
```scala
val fields = a ++ 
             b ++ 
             c ..
```

Comment (srachakonda) [reply src/main/scala/com/imaginea/Main.scala]: updated and committed latest changes

File: src/main/scala/com/imaginea/Main.scala [hunk 2]
```diff
@@ -102,36 +103,36 @@ object Main extends App {
...
-      fields ++= stringToJsField(fieldNames(21), i.privateIpAddress)
-      fields ++= stringToJsField(fieldNames(22), i.publicIpAddress)
-      fields ++= stringToJsField(fieldNames(23), i.elasticIP)
-      fields ++= stringToJsField(fieldNames(24), i.monitoring)
-      fields ++= stringToJsField(fieldNames(25), i.rootDeviceType)
-      fields ++= listToJsValue[InstanceBlockDeviceMappingInfo](fieldNames(26), i.blockDeviceMappings, instanceBlockingFormat)
-      fields ++= listToJsValue[SecurityGroupInfo](fieldNames(27), i.securityGroups, securityGroupsFormat)
-      fields ++= List((fieldNames(28), JsBoolean(i.reservedInstance)))
-      fields ++= stringToJsField(fieldNames(29), i.region)
+      fields ++= longToJsField(fieldNames.head, i.id) ++
```
Comment (eklavya) [inline src/main/scala/com/imaginea/Main.scala]: You should read more carefully. Why not
```scala
val fields = longToJsField(fieldNames.head, i.id) ++
```
Why does it have to be a mutable collection?

Comment (srachakonda) [reply src/main/scala/com/imaginea/Main.scala]: yeah updating it

File: src/main/scala/com/imaginea/activegrid/core/utils/ActiveGridUtils.scala [hunk 3]
```diff
@@ -34,6 +35,64 @@ object ActiveGridUtils {
...
+      }
+    } else {
+      None
+    }
+  }
+
+  def getObjectsFromJson[T: Manifest](propertyMap: Map[String, JsValue], property: String, formateObject: RootJsonFormat[T]): List[T] = {
+    if (propertyMap.contains(property)) {
+      val listOfObjs = propertyMap(property).asInstanceOf[JsArray]
+      listOfObjs.elements.foldLeft(List[T]()) {
```
Comment (eklavya) [inline src/main/scala/com/imaginea/activegrid/core/utils/ActiveGridUtils.scala]: Why is this not a map?

File: src/main/scala/com/imaginea/activegrid/core/utils/ActiveGridUtils.scala [hunk 4]
```diff
@@ -52,9 +52,8 @@ object ActiveGridUtils {
   def getObjectsFromJson[T: Manifest](propertyMap: Map[String, JsValue], property: String, formateObject: RootJsonFormat[T]): List[T] = {
     if (propertyMap.contains(property)) {
       val listOfObjs = propertyMap(property).asInstanceOf[JsArray]
-      listOfObjs.elements.foldLeft(List[T]()) {
-        (list, jsString) =>
-          formateObject.read(jsString) :: list
+      listOfObjs.elements.toList.map{
```
Comment (eklavya) [inline src/main/scala/com/imaginea/activegrid/core/utils/ActiveGridUtils.scala]: It can be
```scala
listOfObjs.elements.toList.map(formateObject.read)
```

**Pull request merged**